.github/workflows/ubuntu-windows.yml

```diff
@@ -0,0 +1,363 @@
+            dotnet: OFF
+          - extras: OFF
+            java: OFF
+            python: OFF
+            dotnet: OFF
+
+
+    steps:
+      - name: set name variables
+        id: names
+        shell: bash
+        run: |
+          REF=${GITHUB_REF#refs/*/}
+          REF_WO_SLASHES=${REF//\//-}
+          WITH_VERSION="_${REF_WO_SLASHES}"
+          SHARED=${{ matrix.shared }}
+          [ $SHARED == "ON" ] && WITH_SHARED="_shared" || WITH_SHARED="_static"
+          XPRS=${{ matrix.xprs }}
+          [ $XPRS == "ON" ] && WITH_XPRS="_xprs" || WITH_XPRS=""
+          SIRIUS=${{ matrix.sirius }}
+          [ $SIRIUS == "ON" ] && WITH_SIRIUS="_sirius" || WITH_SIRIUS=""
+          OS="_${{ matrix.os }}"
+          APPENDIX="${OS}${WITH_SHARED}${WITH_SIRIUS}${WITH_XPRS}"
+          APPENDIX_VERSION="${OS}${WITH_SHARED}${WITH_SIRIUS}${WITH_XPRS}${WITH_VERSION}"
+          echo "::set-output name=appendix::$APPENDIX"
+          echo "::set-output name=appendix_version::$APPENDIX_VERSION"
+
+      - name: Get release
+        if: github.event_name == 'release' && github.event.action == 'created'
+        id: get_release
+        uses:
+          bruceadams/get-release@v1.2.3
+
+      - name: install zip for windows
+        if: ${{ matrix.os == 'windows-2019'}}
+        shell: cmd
+        run: |
+          choco install zip --no-progress
+
+      - name: Install SWIG 4.0.2 for windows
+        if: ${{ matrix.os == 'windows-2019'}}
+        run: |
+          (New-Object System.Net.WebClient).DownloadFile("http://prdownloads.sourceforge.net/swig/swigwin-4.0.2.zip","swigwin-4.0.2.zip");
+          Expand-Archive .\swigwin-4.0.2.zip .;
+          echo "$((Get-Item .).FullName)/swigwin-4.0.2" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+          Remove-Item .\swigwin-4.0.2.zip
+
+      - name: Checkout or-tools
+        uses: actions/checkout@v2
+
+      - name: ccache
+        if: ${{ matrix.os != 'windows-2019'}}
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ matrix.os }}-${{ matrix.shared }}-${{ matrix.sirius }}-${{ matrix.xprs }}
+
+      - name: setup envoronment
+        id: setup
+        uses: ./.github/workflows/setup-env-ubuntu
+        with:
+          sirius-ref: ${{ env.Sirius_Ref }}
+          python-version: '3.10'
+          dotnet: ${{ matrix.dotnet }}
+          python: ${{ matrix.python }}
+
+      - name: Upload sirius install
+        if: ${{ matrix.xprs == 'ON' && matrix.sirius == 'ON' && matrix.shared == 'ON' }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.os }}_${{ steps.setup.outputs.sirius-archive-name }}
+          path: ${{ steps.setup.outputs.sirius-archive-path }}
+
+      - name: Publish sirius install
+        if:  ${{ github.event_name == 'release' && github.event.action == 'created' && matrix.extras == 'ON' && matrix.xprs == 'ON' && matrix.sirius == 'ON' }}
+        uses: actions/upload-release-asset@v1.0.2
```

Comment:
```suggestion
      - name: Publish sirius install
        if:  ${{ github.event_name == 'release' && github.event.action == 'created' && matrix.upload_sirius == 'ON' && matrix.xprs == 'ON' && matrix.sirius == 'ON' }}
        uses: actions/upload-release-asset@v1.0.2
```

Reply:
you are right. I changed strategy, I have a separate job (within the same workflow) that prepares the sirius installation only once and post it as an artifact